src/components/tabs/Tabbable.js

```diff
@@ -0,0 +1,71 @@
+const PropTypes = require('prop-types');
+const React = require('react');
+
+const StyleConstants = require('../../constants/Style');
+
+/**
+ * Common state and prop management and outer styling for Tabs components.
+ */
+const Tabbable = (TabsComponent) => {
+  return class extends React.Component { // eslint-disable-line react/display-name
+    static propTypes = Tabbable.propTypes;
+
+    static defaultProps = {
+      alignment: 'left',
+      selectedTab: 0,
+      showBottomBorder: true
+    };
+
+    constructor (props) {
+      super(props);
+
+      this.state = {
+        selectedTab: this.props.selectedTab
+      };
+    }
+
+    componentWillReceiveProps (nextProps) {
+      if (nextProps.selectedTab !== this.state.selectedTab) {
+        this.setState({
+          selectedTab: nextProps.selectedTab
+        });
+      }
+    }
+
+    handleTabSelect (selectedTab) {
+      this.props.onTabSelect(selectedTab);
+
+      this.setState({
+        selectedTab
+      });
+    }
```

Comment:
Good catch! I was supposed to move https://github.com/mxenabled/mx-react-components/pull/606/files#diff-6d37b6931020e71a211c578f514954b6L218 to `render` in this file.